### ISSUE TITLE
Rule Change: (space-in-parens)

### ISSUE THREAD

**Issue by mlwyatt** [eslint/eslint#16118]
### What rule do you want to change?

space-in-parens

### What change to do you want to make?

Implement suggestions

### How do you think the change should be implemented?

A new option

### Example code

```js
# Always

/* eslint space-in-parens: ["error", "always"] */
if ( Object.prototype.hasOwnProperty.call(obj, [el]) ) {
  objSum += obj[el];
}


# Never
```js
/* eslint space-in-parens: ["error", "never"] */
if ( Object.prototype.hasOwnProperty.call(obj, [el]) ) {
  objSum += obj[el];
}
```
```


### What does the rule currently do for this code?

# Always
```txt
  2:42  error  There must be a space after this paren   space-in-parens
  2:52  error  There must be a space before this paren  space-in-parens
```
Fixed:
```js
if ( Object.prototype.hasOwnProperty.call( obj, [el] ) ) {
  objSum += obj[el];
}
```

# Never
```txt
  2:5   error  There should be no space after this paren   space-in-parens
  2:53  error  There should be no space before this paren  space-in-parens
```
Fixed:
```js
if (Object.prototype.hasOwnProperty.call(obj, [el])) {
  objSum += obj[el];
}
```

### What will the rule do after it's changed?

What I would like to achieve is requiring spaces within conditionals but not function calls so that the following code is valid.

```js
if ( Object.prototype.hasOwnProperty.call(obj, [el]) ) {
  objSum += obj[el];
}
```

### Participation

- [ ] I am willing to submit a pull request to implement this change.

### Additional comments

I know it was mentioned in #15100 that `space-in-parens` and `func-call-spacing` combined would solve this but that's incorrect. `func-call-spacing` is for requiring/disallowing spaces between the function name and the calling `()`, not within the `()`

**Comment by mlwyatt** [comment]
Also similar is #14102

**Comment by mdjermanovic** [comment]
Hi @mlwyatt, thanks for the issue!

Per our rules policies, [stylistic rules are frozen](https://github.com/eslint/eslint#stylistic-rule-updates), so we are not adding new options to the `space-in-parens` rule. You can always copy the existing rule into a custom rule and make it behave the way you want.
